Feat: PKI Certs for CP/DP authentication (#6155)
Kong/docs.konghq.com: app/konnect/gateway-manager/data-plane-nodes/renew-certificates.md

---
title: Authentication Certificates for a Data Plane Node
content_type: how-to
---

{{site.konnect_short_name}} uses a mutual TLS handshake (mTLS) for authentication between data plane and control plane so the actual private key is never transferred on the network, and communication between the control plane and data plane nodes is secure. {{site.konnect_short_name}} supports two modes for handling certificate/key pairs:
* **Pinned mode**: This is the default mode. The same public key is added to the data plane and the control plane, and the control plane uses this public key to authenticate the data plane. Validation of the certificate only happens on the control plane.

> * **Public Key Infrastructure (PKI) mode**: This mode uses digital certificates signed by a certificate authority, or a chain of certificate authorities, to authenticate between control plane and data plane. The public key is added to the data plane, while the chain of certificate authority is added to the control plane. Any certificate authority from the chain can be used to authenticate between dataplane and control plane.  {{site.konnect_short_name}} validates both the control plane and data plane sides by checking if they are from the same certificate authority, thereby increasing the security of the network and eliminating the risks associated with transporting private keys. This mode is only supoprted for Control Plane and Control Plane Groups.

Author: ```suggestion
* **Public Key Infrastructure (PKI) mode**: This mode uses digital certificates signed by a certificate authority, or a chain of certificate authorities, to authenticate between control plane and data plane. The public key is added to the data plane, while the chain of certificate authority is added to the control plane. Any certificate authority from the chain can be used to authenticate between dataplane and control plane.  {{site.konnect_short_name}} validates both the control plane and data plane sides by checking if they are from the same certificate authority, thereby increasing the security of the network and eliminating the risks associated with transporting private keys. This mode is only supported for Control Plane and Control Plane Groups.
```

> ## Set certificate authentication mode on the Control Plane

Author: ```suggestion
## Set certificate authentication mode
```

Consider the following scenarios with this example cert chain:
   * `cert1`: service (issuer: intermediary)
   * `cert2`: intermediary (issuer: root)
   * `cert3`: root (issuer: root / self signed)

* **Upload only cert1 to the control plane**: This is the Pinned mode. You can include just `cert1` in your data plane request and not include the chain. The control plane doesn’t need to evaluate the issuer because it trusts the cert itself.
* **Upload only cert2 to the control plane**: This would mean any cert coming in that has (issuer: intermediary) would be trusted. You can include just `cert1` in your data plane request. The control plane would trust any certificate issued by the intermediary public key. 
* **Upload only cert3 to the control plane**: This is the typical PKI case. It means any cert signed by the root is trusted. However, since `cert1` is signed by an intermediary and `cert2` is signed by root, you need to include both `cert1` and `cert2` in your data plane request. The control plane would trust the whole chain because `cert2` is issued by `cert3` and `cert1` is issued by `cert2`.

> You can generate pinned certificates in {{site.konnect_saas}} or bring your own pinned and PKI certificates. Data plane certificates generated by {{site.konnect_saas}} expire every ten years. If you bring your own certificates, make sure to review the expiration date and associated metadata.

Author: ```suggestion
You can generate pinned certificates in {{site.konnect_short_name}} or bring your own pinned and PKI certificates. Data plane certificates generated by {{site.konnect_short_name}} expire every ten years. If you bring your own certificates, make sure to review the expiration date and associated metadata.
```

> ## Renew Data Plane Node Certificates

Author: ```suggestion
## Renew certificates
```
